agenta-cli/sdk.py

```diff
@@ -0,0 +1,652 @@
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel
+
+
+class Reference(BaseModel):
+    # IDENTIFIER
+    id: Optional[str] = None
+    slug: Optional[str] = None
+    # VERSION
+    version: Optional[str] = None
+
+
+class Application(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+
+
+class Service(BaseModel):
+    # INTERNALS
+    url: str
+
+
+class Configuration(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
+    # HEADER
+    name: Optional[str] = None
+    description: Optional[str] = None
+    # VERSION
+    version: str
+    latest: bool
+
+    # REFERENCES
+    application: Application
+
+    # INTERNALS
+    parameters: Dict[str, Any]
+
+
+class Variant(BaseModel):
+    # SCOPE
+    project_id: str
+    # IDENTIFIER
+    id: str
+    slug: str
```

Comment:
How should we identify the variant in app structure 2 (assuming no large migration)? What does that identifier points to? The last version of the variant?

identifier -> last version of the variant
Read(identifier) -> returns last version of variant
Write(identifier) -> Creates a new version of the variant and sets it to latest 
Creates(identifier) -> Creates a new variant based on the default base_id 
Deploy(identifier, environment) -> deploy the last version of the variant to the environment (on the background we deploy instead the variant revision and not the variant itself)

In this case, we might be able to use the slug as the identifier, no? (also easily add stuff like description)

Reply:
```
**_Reference_**

_Eventually_ -- after _Application Structure_
(commit) `id`, or
`slug` + `version`

_For this RFC_ -- without any refactoring
`variant_revision:id` (~ commit id)
`variant:id` (~ slug) + `variant_revision:revision` (~ version)
```